### PR TITLE
Add finalizer for storage-protection

### PR DESCRIPTION
**Pull request by vatsa287**
Signed-off-by: Shree Vatsa N <vatsa@kadalu.io>

### CODE REVIEW

**Comment by vatsa287** [comment]
Any reason why the same command varies from operator and on normal terminal?
```bash
root@operator-7dfcc8fb86-4554c:/# kubectl get kadalustorages.kadalu-operator.storage -oyaml
apiVersion: v1
items: []
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```
&
```
kubectl get kadalustorages.kadalu-operator.storage -oyaml
apiVersion: v1
items:
- apiVersion: kadalu-operator.storage/v1alpha1
  kind: KadaluStorage
  metadata:
    annotations:
      kubectl.kubernetes.io/last-applied-configuration: |
        {"apiVersion":"kadalu-operator.storage/v1alpha1","kind":"KadaluStorage","metadata":{"annotations":{},"finalizers":["storage-protection"],"name":"storage-pool-1","namespace":"default"},"spec":{"storage":[{"node":"thinkpad","path":"/mnt/disk1"},{"node":"thinkpad","path":"/mnt/disk2"},{"node":"thinkpad","path":"/mnt/disk3"}],"type":"Replica3"}}
    creationTimestamp: "2021-12-22T14:59:22Z"
    finalizers:
    - storage-protection
    generation: 1
    name: storage-pool-1
    namespace: default
    resourceVersion: "5220"
    uid: 2f2d1af3-4c2a-47a3-95e0-a53a070dae98
  spec:
    kadalu_format: native
    pvReclaimPolicy: delete
    storage:
    - node: thinkpad
      path: /mnt/disk1
    - node: thinkpad
      path: /mnt/disk2
    - node: thinkpad
      path: /mnt/disk3
    type: Replica3
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```
Is it because `-nkadalu` is pre-embedded inside operator. By changing `scope: Cluster` in crd template could not add storage. Is any new rules needed to be added?

File: operator/main.py [hunk 1]
```diff
@@ -650,20 +663,27 @@ def handle_deleted(core_v1_client, obj):
...
-            delete_server_pods(storage_info_data, obj)
             filename = os.path.join(MANIFESTS_DIR, "services.yaml")
-            template(filename, namespace=NAMESPACE, volname=volname)
+            with open(filename, 'r') as f:
+                logging.info(logf(
+                    "services.yaml",
+                    services=f.read()
+                ))
+            resource_to_be_patched = "kadalustorages.kadalu-operator.storage/" + volname
+            lib_execute(KUBECTL_CMD, PATCH_CMD, resource_to_be_patched, "-p",
```
Comment (leelavg) [inline operator/main.py:675]: - where is the finalizer being set in the first place?

Comment (vatsa287) [reply operator/main.py:675]: The finalizer is being set in storage_config right now through user end[sample-storage.yaml]. I tried to set it by default in crd templates/operator.j2 but it is not being set.

**Comment by leelavg** [comment]
@vatsa287 
- this PR can be dropped, there are some expectations how operators should work and ours is simple according to our needs.
- once we merge our efforts with Moana then I'll try to persuade @aravindavk to use [kopf](https://github.com/nolar/kopf) 😅 
- pardon the inconvenience

**Comment by vatsa287** [comment]
Thanks. Will close this for now.

**Pull request closed without merging**